homeassistant/components/slide/cover.py

```diff
@@ -0,0 +1,128 @@
+"""Support for Go Slide slides."""
+
+import logging
+
+from homeassistant.util import slugify
+from homeassistant.components.cover import (
+    ATTR_POSITION,
+    ENTITY_ID_FORMAT,
+    SUPPORT_CLOSE,
+    SUPPORT_OPEN,
+    SUPPORT_SET_POSITION,
+    SUPPORT_STOP,
+    STATE_OPEN,
+    STATE_CLOSED,
+    DEVICE_CLASS_CURTAIN,
+    CoverDevice,
+)
+from .const import API, DOMAIN, SLIDES
+
+_LOGGER = logging.getLogger(__name__)
+
+
+# pylint: disable=unused-argument
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up cover(s) for Go Slide platform."""
+    entities = []
+
+    for key in hass.data[DOMAIN][SLIDES]:
+        _LOGGER.debug("Setting up Slide entity: %s", hass.data[DOMAIN][SLIDES][key])
+        entities.append(SlideCover(hass, hass.data[DOMAIN][SLIDES][key]))
+
+    async_add_entities(entities)
+
+
+class SlideCover(CoverDevice):
+    """Representation of a Go Slide cover."""
+
+    def __init__(self, hass, slide):
+        """Initialize the cover."""
+        self._hass = hass
+        self._mac = slide["mac"]
+        self._id = slide["id"]
+        self._name = slide["name"]
+        self._entity_id = ENTITY_ID_FORMAT.format(slugify("slide_" + self._mac))
```

Comment:
We can use the mac address as a unique id. Then the user can change entity_id and name via GUI. Please implement the `unique_id` entity property:
https://developers.home-assistant.io/docs/en/entity_index.html#generic-properties
https://developers.home-assistant.io/docs/en/entity_registry_index.html#unique-id-requirements
```suggestion
```

Reply:
Made the change, and it is a good improvement - thanks 👍